pkg/retry/retry_test.go

```diff
@@ -439,3 +440,48 @@ func TestOnError_NegativeRetryLimit(t *testing.T) {
 		require.Contains(t, line, expectedErrMsg)
 	}
 }
+
+// TestCallWithExponentialBackoff tests the Call function with exponential backoff
+func TestCallWithExponentialBackoff(t *testing.T) {
+	t.Run("succeeds after retries", func(t *testing.T) {
+		attempts := 0
+		maxAttempts := 3
+		ctx := context.Background()
+
+		result, err := retry.Call(
+			ctx,
+			func() (string, error) {
+				attempts++
+				if attempts < maxAttempts {
+					return "", errors.New("not yet")
+				}
+				return "success", nil
+			},
+			retry.WithExponentialBackoffFn(5, 1, 10), // Very short delays for testing
```

Comment:
Make these local vars for the test. 

Makes it easier to understand what we're dealing with / testing.